### PR TITLE
Microsoft Search: Spelling suggestions interface

### PR DESCRIPTION
**Pull request by edarroudi**
I added the spelling suggestions Interface with options to set through property pane. Extended the response so it can be used in the search result webparts.

I don't know if it's mandatory to add the new labels to the different languages ( i just copied the english label to the language files for now).

### CODE REVIEW

**Comment by edarroudi** [comment]
Should i try to get past this stupid Duplication message? Of course there is some kind of duplication for labels.

**Comment by edarroudi** [comment]
Hi,

i updated the code for the develop merge and added a check for graph beta. Should i do something else here?

**Comment by wobba** [comment]
Wait :)

**Comment by wobba** [comment]
@FranckyC do you see any issues with merging this one?

File: search-parts/src/dataSources/MicrosoftSearchDataSource.ts [hunk 1]
```diff
@@ -618,9 +648,17 @@ export class MicrosoftSearchDataSource extends BaseDataSource<IMicrosoftSearchDa
         const msGraphClientFactory = this.serviceScope.consume<MSGraphClientFactory>(MSGraphClientFactory.serviceKey);
         const msGraphClient = await msGraphClientFactory.getClient();
         const request = await msGraphClient.api(this._microsoftSearchUrl);     
-            
-        const jsonResponse = await request.post({ requests: [searchRequest] });
 
+        let jsonResponse = undefined;
+
+        if (this.properties.useBetaEndpoint &&
```
Comment (FranckyC) [inline search-parts/src/dataSources/MicrosoftSearchDataSource.ts:654]: There properties should be added in the `IMicrosoftSearchRequests` through a top level interface in this case

File: search-parts/src/dataSources/MicrosoftSearchDataSource.ts [hunk 2]
```diff
@@ -344,7 +350,29 @@ export class MicrosoftSearchDataSource extends BaseDataSource<IMicrosoftSearchDa
             }));
         } 
 
-        return  [
+        if (this.properties.useBetaEndpoint  &&
```
Comment (FranckyC) [inline search-parts/src/dataSources/MicrosoftSearchDataSource.ts:353]: The condition does not work as I can see these options even the "Use beta endpoint" flag is disabled.

image

File: search-parts/src/models/search/IAlteredQueryTokens.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export interface IAlteredQueryTokens {
```
Comment (FranckyC) [inline search-parts/src/models/search/IAlteredQueryTokens.ts:1]: As they are specific to Microsoft Search, these interface should be in the `IMicrosoftSearchResponse.ts` file and included in the response schema

Comment (FranckyC) [reply search-parts/src/models/search/IAlteredQueryTokens.ts:1]: + add links to Microsoft Search API reference

File: search-parts/src/dataSources/MicrosoftSearchDataSource.ts [hunk 4]
```diff
@@ -16,6 +16,8 @@ import { AsyncCombo } from "../controls/PropertyPaneAsyncCombo/components/AsyncC
 import { IAsyncComboProps } from "../controls/PropertyPaneAsyncCombo/components/IAsyncComboProps";
 import { ISharePointSearchService } from "../services/searchService/ISharePointSearchService";
 import { SharePointSearchService } from "../services/searchService/SharePointSearchService";
+import {IMicrosoftSearchDataSourceData} from "../models/search/IMicrosoftSearchDataSourceData";
```
Comment (FranckyC) [inline search-parts/src/dataSources/MicrosoftSearchDataSource.ts:19]: Spaces here according to the other imports.

**Review by FranckyC — CHANGES_REQUESTED** [review]
Please update according comments before merging Thanks!

**Comment by FranckyC** [comment]
Hi @edarroudi, changes have been integrated with PR #1622. Needed for December release. I'll close this one.

**Pull request closed without merging**